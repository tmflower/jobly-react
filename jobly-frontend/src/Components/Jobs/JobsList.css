@media (min-width: 280px) {
    .balloon4, .balloon6 {
        display: none;
    }

}

@media (min-width: 768px) {
    .balloon4 {
        display: block;
        position: fixed;
        width: 15%;
        left: 4%;
        z-index: -1;
        transform: rotate(-10deg);
        top: 12%;
    }

    .balloon6 {
        display: block;
        position: fixed;
        width: 15%;
        right: 1%;
        z-index: -1;
        transform: rotate(20deg);
        top: 10%;
    }
}