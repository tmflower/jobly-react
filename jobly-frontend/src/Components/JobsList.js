import React, { useState, useEffect } from "react";
import JoblyApi from "../api";
import JobCard from "./JobCard";

const JobsList = () => {

    // sets initial state of jobs to empty array; will be filled with data from api;
    const [jobs, setJobs] = useState([]);
    const [searchTerm, setSearchTerm] = useState('');

    // requests data for all jobs from api and saves in state
    useEffect(function getAllJobs() {
        async function getJobs() {
            let allJobs = await JoblyApi.getJobs();
            setJobs(allJobs);
        }
        getJobs();
    }, []);

    // when search terms are entered, requests data from api on jobs that contain the terms and saves in state
    async function searchJobs(searchTerm) {
        let selectedJobs = await JoblyApi.getJobs(searchTerm);
        setJobs(selectedJobs);
        setSearchTerm('');
}

    // gets the user's search terms from the form and saves in state
    const handleChange = (evt) => {
        setSearchTerm(evt.target.value);
    }

    // when search terms are entered and button is clicked, this calls the function to get the filtered results
    const handleSubmit = (evt) => {
        evt.preventDefault();
        searchJobs(searchTerm);        
    }

    // displays a list of all the jobs displaying the title, company, salary, and equity for each job
    // displays a search bar allowing user to filter jobs by job title
    return (
        <div>
            <h1>Jobs</h1>
            <form onSubmit={handleSubmit}>
                <label htmlFor="searchTerm">
                <input id="searchTerm" type="text" placeholder="Enter search terms" name="searchTerm" value={searchTerm} onChange={handleChange}/></label>
                <button>Submit</button> 
            </form> 
            <JobCard />
            {jobs.map(job => (<JobCard title={job.title} company={job.companyName} salary={job.salary} equity={job.equity} key={job.id}/>))}
        </div>
    )
}

export default JobsList;