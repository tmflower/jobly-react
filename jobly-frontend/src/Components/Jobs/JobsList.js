import React, { useState, useEffect } from "react";
import JoblyApi from "../../api.js";
import JobCard from "./JobCard";
import userContext from "../Layout/userContext";
import { useNavigate } from "react-router-dom";
import balloon3 from "../Images/balloon3.jpg";
import balloon5 from "../Images/balloon5.jpg";
import "./JobsList.css";

const JobsList = ({ applications, addJob }) => {

    // provides current username
    const username = React.useContext(userContext);

    // sets initial state of jobs to empty array; will be filled with data from api;
    const [jobs, setJobs] = useState([]);
    const [searchTerm, setSearchTerm] = useState('');

    // requests data for all jobs from api and saves in state
    useEffect(function getAllJobs() {
        async function getJobs() {
            let allJobs = await JoblyApi.getJobs();
            setJobs(allJobs);
        }
        getJobs();
    }, []);

    // when search terms are entered, requests data from api on jobs that contain the terms and saves in state
    async function searchJobs(searchTerm) {
        let selectedJobs = await JoblyApi.getJobs(searchTerm);
        setJobs(selectedJobs);
        setSearchTerm('');
}

    // gets the user's search terms from the form and saves in state
    const handleChange = (evt) => {
        setSearchTerm(evt.target.value);
    }

    // when search terms are entered and button is clicked, this calls the function to get the filtered results
    const handleSubmit = (evt) => {
        evt.preventDefault();
        searchJobs(searchTerm);        
    }

    // provides onClick function for button that redirects user to log in, if not logged in already
    const navigate = useNavigate();
    const forceLogin = () => {
        navigate("/login", { replace: true });
    }

    // if no logged-in user, jobs are hidden and user is prompted to login
    // if user is logged in:
    // displays a list of all the jobs displaying the title, company, salary, and equity for each job
    // displays a search bar allowing user to filter jobs by job title
    return (
        <div>
            <img src={balloon3} className="balloon4" alt="yellow hot air balloon with girl"/>
            <h3>Check out our current job openings!</h3>
            <img src={balloon5} className="balloon6" alt="yellow hot air balloon with girl"/>
            { username === null ? <div><h3>You must log in to see our jobs.</h3><button onClick={forceLogin}>Log In</button></div> :
            <div>
                <form onSubmit={handleSubmit}>
                    <label htmlFor="searchTerm">Search for a job
                    <input 
                        id="searchTerm" 
                        type="text" 
                        placeholder="Enter job title" 
                        name="searchTerm" 
                        value={searchTerm} 
                        onChange={handleChange}/></label>
                    <button>Submit</button> 
                </form> 
                {jobs.map(job => (<JobCard 
                                    id={job.id}
                                    title={job.title} 
                                    company={job.companyName} 
                                    salary={job.salary} 
                                    equity={job.equity} 
                                    key={job.id}
                                    applications={applications}
                                    addJob={addJob} />))}
            </div>
            }
        </div>
    )
}

export default JobsList;