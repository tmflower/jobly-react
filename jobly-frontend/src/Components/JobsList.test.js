import React from "react";
import { render } from "@testing-library/react";
import JobsList from "./JobsList";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", function() {
  render(<MemoryRouter><JobsList /></MemoryRouter>);
});

it("matches snapshot", function () {
    const { asFragment } = render(<MemoryRouter><JobsList /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });