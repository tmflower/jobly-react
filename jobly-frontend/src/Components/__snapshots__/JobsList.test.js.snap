// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Jobs
    </h1>
    <div>
      <form>
        <label
          for="searchTerm"
        >
          <input
            id="searchTerm"
            name="searchTerm"
            placeholder="Enter search terms"
            type="text"
            value=""
          />
        </label>
        <button>
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
